app/src/main/java/com/example/wizeline/ui/adapters/ListCoinsAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.wizeline.ui.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.wizeline.R
+import com.example.wizeline.data.datasource.models.BookInfoEntity
+import com.example.wizeline.databinding.ItemCurrencyBinding
+
+class ListCoinsAdapter(private val action: (item: BookInfoEntity) -> Unit) :
+    ListAdapter<BookInfoEntity, ListCoinsAdapter.ViewHolder>(compare) {
+
+    companion object{
+        val compare = object: DiffUtil.ItemCallback<BookInfoEntity>(){
+            override fun areItemsTheSame(oldItem: BookInfoEntity, newItem: BookInfoEntity): Boolean {
+                return oldItem.book == newItem.book
+            }
+
+            override fun areContentsTheSame(oldItem: BookInfoEntity, newItem: BookInfoEntity): Boolean {
+                return oldItem == newItem
+            }
+
+        }
+    }
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
+        ViewHolder(ItemCurrencyBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bindItem(getItem(position))
+    }
+
+    inner class ViewHolder(private val itemBinding: ItemCurrencyBinding) : RecyclerView.ViewHolder(itemBinding.root) {
+
+        fun bindItem(item: BookInfoEntity) = with(itemBinding) {
+            tvItem.text = item.book
+            setImageCoin(ivCurrency, item.book!!)
```

Comment:
recuerda no utilizar el !! ya que puede provocar `NullPointerException`

```suggestion
            item?.let{
                        setImageCoin(ivCurrency, item.book)
            }
```